src/dhcp/aca_dhcp_state_handler.cpp

```diff
@@ -0,0 +1,125 @@
+// Copyright 2019 The Alcor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "aca_log.h"
+#include "aca_dhcp_server.h"
+#include "aca_dhcp_state_handler.h"
+#include "goalstateprovisioner.grpc.pb.h"
+#include <future>
+
+using namespace alcorcontroller;
+using namespace aca_dhcp_programming_if;
+
+namespace aca_dhcp_state_handler
+{
+Aca_Dhcp_State_Handler::Aca_Dhcp_State_Handler()
+{
+  ACA_LOG_INFO("DHCP State Handler initialize\n");
+
+  // default to dataplane_mizar for now
+  ACA_LOG_INFO("DHCP State Handler: using dhcp server\n");
+  this->dhcp_programming_if = new aca_dhcp_server::ACA_Dhcp_Server;
+
+  int rc = this->dhcp_programming_if->initialize();
+
+  if (rc == EXIT_SUCCESS) {
+    ACA_LOG_INFO("DHCP Programming initialization succeed\n");
+  } else {
+    ACA_LOG_ERROR("DHCP Programming initialization failed\n");
+    throw std::system_error(ENXIO, std::generic_category(),
+                            "DHCP Programming initialization failed\n");
+  }
+}
+
+Aca_Dhcp_State_Handler::~Aca_Dhcp_State_Handler()
+{
+  // allocated dhcp_programming_if is destroyed when program exits.
+}
+
+Aca_Dhcp_State_Handler &Aca_Dhcp_State_Handler::get_instance()
+{
+  // It is instantiated on first use.
+  // allocated instance is destroyed when program exits.
+  static Aca_Dhcp_State_Handler instance;
+  return instance;
+}
+
+int Aca_Dhcp_State_Handler::update_dhcp_state_workitem(const DhcpState current_DhcpState)
```

Comment:
we will need to update this function to also take in GoalStateOperationReply &gsOperationReply as an input parameter, and then update the gsOperationReply at the end. Please see the below for reference:

int ACA_Dataplane_OVS::update_subnet_state_workitem(const SubnetState current_SubnetState,
                                                    GoalStateOperationReply &gsOperationReply)